### PR TITLE
Add Smarter handling for Recipes within the alloy forge

### PR DESCRIPTION
**Pull request by Dragon-Seeker**
yeet

### CODE REVIEW

File: src/main/java/wraith/alloyforgery/recipe/AlloyForgeRecipeSerializer.java [hunk 1]
```diff
@@ -19,10 +24,25 @@ public class AlloyForgeRecipeSerializer implements RecipeSerializer<AlloyForgeRe
...
+        Map<JsonObject, Integer> jsonObjectIntegerMap = new HashMap<>();
 
-        JsonHelper.getArray(json, "inputs").forEach(jsonElement -> inputs.add(Ingredient.fromJson(jsonElement)));
-        if (inputs.isEmpty()) throw new JsonSyntaxException("Inputs cannot be empty");
+        for(JsonElement entry : JsonHelper.getArray(json, "inputs")){
+            JsonObject object = entry.getAsJsonObject();
+
+            if(jsonObjectIntegerMap.containsKey(object)){
+                jsonObjectIntegerMap.replace(object, jsonObjectIntegerMap.get(object) + 1);
+            }else{
```
Comment (gliscowo) [inline src/main/java/wraith/alloyforgery/recipe/AlloyForgeRecipeSerializer.java:34]: mfw

Comment (Dragon-Seeker) [reply src/main/java/wraith/alloyforgery/recipe/AlloyForgeRecipeSerializer.java]: fu

**Comment by Dragon-Seeker** [comment]
Should be pretty close to done

**Comment by Dragon-Seeker** [comment]
Now it is done.

**Review by gliscowo — APPROVED** [review]
podge moment, very cool

**Pull request merged**